assets/js/components/settings/SettingsActiveModule/Header.js

```diff
@@ -68,6 +72,21 @@ export default function Header( { slug } ) {
 	const isGA4Connected = useSelect( ( select ) =>
 		select( CORE_MODULES ).isModuleConnected( 'analytics-4' )
 	);
+	const loggedInUserID = useSelect( ( select ) =>
+		select( CORE_USER ).getID()
+	);
+	const hasAnalyticsAccess = useSelect( ( select ) => {
+		if ( ! ( slug === 'analytics' && module?.connected ) ) {
+			return false;
+		}
+
+		const moduleOwnerID = select( MODULES_ANALYTICS ).getOwnerID();
+
+		if ( moduleOwnerID === loggedInUserID ) {
+			return true;
+		}
+		return select( CORE_MODULES ).hasModuleAccess( 'analytics' );
+	} );
```

Comment:
@aaemnnosttv We had added a new [selector](https://github.com/google/site-kit-wp/pull/5994/files#diff-51079bbb133e20fa794c0e1fd5fd30b551dda818e0cfe0ba4162502670cf133aR1241-R1269) to avoid this chunk being constantly copy pasted everywhere when checking for the current user's access to a module - and the chunk that was used in the `ReminderBanner` component. The plan was to eventually refactor the existing copy-pasted code with the new selector too. Any reason that selector was [removed](https://github.com/google/site-kit-wp/commit/74dbb3a2c70d0d2965642250414f6fdd7ac4a43d)?

c.c. @hussain-t

Reply:
@jimmymadon there was a short explanation on the PR it was removed in here: https://github.com/google/site-kit-wp/pull/6046/files#r1001939812

The other reason was it was a bit unclear in the naming/docs between that one and `hasModuleAccess` which is also specific to the current user. So I think there is a bit more thought to put into this. That may be a new selector, a change to `hasModuleAccess` or something else, but now that we have more than one instance, let's open a new issue to explore the right solution there.

One other detail about the removed selector was around unexpected inconsistency with "has" semantics which returned an object rather than a boolean as all other "has" selectors do. I generally like the idea of a React-query inspired API (which I think was the intention here) but this should probably be implemented in a more reusable way. See [`useQuerySelect`](https://github.com/WordPress/gutenberg/blob/7822ef818daeb267e8f675fda8d901eca376bc00/packages/core-data/src/hooks/use-query-select.ts) from core-data for an example :)

Reply:
@aaemnnosttv Thanks for that clarification - makes sense! Created [another issue here](https://github.com/google/site-kit-wp/issues/6192) and wrote a very non-technical AC (so that we can leave those discussions to the IB)